firmware-support/bittide-sys/src/smoltcp/loopback.rs

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: 2024 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+use heapless::{Deque, Vec};
+
+use smoltcp::phy::{self, Device, DeviceCapabilities, Medium};
+use smoltcp::time::Instant;
+
+/// A loopback device.
+pub struct Loopback {
+    pub(crate) queue: Deque<Vec<u8, 1024>, 8>,
+    medium: Medium,
+}
+
+#[allow(clippy::new_without_default)]
+impl Loopback {
+    /// Creates a loopback device.
+    ///
+    /// Every packet transmitted through this device will be received through it
+    /// in FIFO order.
+    pub fn new(medium: Medium) -> Loopback {
+        Loopback {
+            queue: Deque::new(),
+            medium,
+        }
+    }
+}
+
+impl Device for Loopback {
+    type RxToken<'a> = RxToken;
+    type TxToken<'a> = TxToken<'a>;
+
+    fn capabilities(&self) -> DeviceCapabilities {
+        let mut cap = DeviceCapabilities::default();
+        cap.max_transmission_unit = 65535;
+        cap.medium = self.medium;
+        cap
+    }
+
+    fn receive(&mut self, _timestamp: Instant) -> Option<(Self::RxToken<'_>, Self::TxToken<'_>)> {
+        self.queue.pop_front().map(move |buffer| {
+            let rx = RxToken { buffer };
+            let tx = TxToken {
+                queue: &mut self.queue,
+            };
+            (rx, tx)
+        })
```

Comment:
What is the `move` moving here? Just `&mut self.queue`? Shouldn't this work without move too?

Reply:
I think I'm going to remove the loopback device as it has outlived its use.
(It was pretty useful for development)